src/main/kotlin/eu/europa/ec/eudi/sdjwt/vc/SdJwtVcVerifier.kt

```diff
@@ -253,14 +257,14 @@ internal fun keySource(jwt: SignedJWT): SdJwtVcIssuerPublicKeySource? {
     }
 
     return when {
-        issScheme == HTTPS_URI_SCHEME && certChain.isEmpty() -> Metadata(issUrl, kid)
-        issScheme == HTTPS_URI_SCHEME -> {
+        issScheme == HTTPS_URI_SCHEME && certChain.isNotEmpty() -> {
             val leaf = certChain.first()
             if (leaf.containsIssuerUri(issUrl) || leaf.containsIssuerDnsName(issUrl)) X509CertChain(issUrl, certChain)
-            else null
+            else invalidSdJwtVc("Failed to find $issUrl to URI or DNS entries of provided leaft certificate")
```

Comment:
@babisRoutis Please rephrase the error as `Failed to find $issUrl in URI or DNS entries of provided leaft certificate`